src/qcodes/dataset/descriptions/dependencies.py

```diff
@@ -50,262 +52,176 @@ def __str__(self) -> str:
+            if edge_data["type"] == type:
+                paramspec_tree_list[self._node_to_paramspec(node_from)].append(
+                    self._node_to_paramspec(node_to)
+                )
+        return {key: tuple(val) for key, val in paramspec_tree_list.items()}
+
+    def _node_to_paramspec(self, node_id: str) -> ParamSpecBase:
+        return cast("ParamSpecBase", self.graph.nodes[node_id]["value"])
+
+    def _paramspec_predecessors_by_type(
+        self, paramspec: ParamSpecBase, type: str
+    ) -> tuple[ParamSpecBase, ...]:
+        return tuple(
+            self._node_to_paramspec(node_from)
+            for node_from, _, edge_data in self.graph.in_edges(
+                paramspec.name, data=True
+            )
+            if edge_data["type"] == type
+        )
 
-        return None
+    @property
+    def dependencies(self) -> ParamSpecTree:
+        return self._paramspec_tree_by_type("depends_on")
 
     def what_depends_on(self, ps: ParamSpecBase) -> tuple[ParamSpecBase, ...]:
-        """
-        Return a tuple of the parameters that depend on the given parameter.
-        Returns an empty tuple if nothing depends on the given parameter
-
-        Args:
-            ps: the parameter to look up
-
-        Raises:
-            ValueError: If the parameter is not part of this object
-
-        """
-        if ps not in self:
-            raise ValueError(f"Unknown parameter: {ps}")
-        return self._dependencies_inv.get(ps, ())
+        return self._paramspec_predecessors_by_type(ps, type="depends_on")
 
     def what_is_inferred_from(self, ps: ParamSpecBase) -> tuple[ParamSpecBase, ...]:
-        """
-        Return a tuple of the parameters thatare inferred from the given
-        parameter. Returns an empty tuple if nothing is inferred from the given
-        parameter
+        return self._paramspec_predecessors_by_type(ps, type="inferred_from")
 
-        Args:
-            ps: the parameter to look up
-
-        Raises:
-            ValueError: If the parameter is not part of this object
-
-        """
-        if ps not in self:
-            raise ValueError(f"Unknown parameter: {ps}")
-        return self._inferences_inv.get(ps, ())
-
-    def _to_dict(self) -> InterDependencies_Dict:
-        """
-        Write out this object as a dictionary
-        """
-        parameters = {
-            key: value._to_dict() for key, value in self._id_to_paramspec.items()
-        }
-        dependencies = self._construct_subdict("dependencies")
-        inferences = self._construct_subdict("inferences")
-        standalones = [self._paramspec_to_id[ps] for ps in self.standalones]
-        output: InterDependencies_Dict = {
-            "parameters": parameters,
-            "dependencies": dependencies,
-            "inferences": inferences,
-            "standalones": standalones,
-        }
-        return output
-
-    def _empty_data_dict(self) -> dict[str, dict[str, npt.NDArray]]:
```

Comment:
This method has not been converted. I can simply copy it as is into the new implementation, but my instinct is that it doesn't belong here in the first place. 

It is only used in the DataSetCache's `prepare` method, and it seems like it would be better to create the empty arrays there than here.

Reply:
Decided to copy/paste as it was to keep the scope of this PR limited.